src/expressions/dataTypes/builtins/dataTypeValidatorByType.ts

```diff
@@ -159,47 +159,41 @@ function isValidDayTimeDuration(value: string): boolean {
 }
 
 export function getValidatorForType(type: ValueType): (value: string) => boolean {
-	return dataTypeValidatorByName[type];
-}
-
-const dataTypeValidatorByName: { [s: string]: (value: string) => boolean } = {
-	'xs:anySimpleType': isValidAnySimpleType,
-	'xs:anyAtomicType': isValidAnyAtomicType,
-
-	'xs:string': isValidString,
-	'xs:boolean': isValidBoolean,
-	'xs:float': isValidFloat,
-	'xs:double': isValidDouble,
-	'xs:decimal': isValidDecimal,
-	'xs:duration': isValidDuration,
-	'xs:dateTime': isValidDateTime,
-	'xs:time': isValidTime,
-	'xs:date': isValidDate,
-	'xs:gYearMonth': isValidGYearMonth,
-	'xs:gYear': isValidGYear,
-	'xs:gMonthDay': isValidGMonthDay,
-	'xs:gDay': isValidGDay,
-	'xs:gMonth': isValidGMonth,
-	'xs:hexBinary': isValidHexBinary,
-	'xs:base64Binary': isValidBase64Binary,
-	'xs:anyURI': isValidAnyURI,
-	'xs:NOTATION': isValidQName,
-
-	'xs:normalizedString': isValidNormalizedString,
-	'xs:token': isValidToken,
-	'xs:language': isValidLanguage,
-	'xs:NMTOKEN': isValidNMTOKEN,
-	'xs:Name': isValidName,
-	'xs:QName': isValidQName,
-	'xs:NCName': isValidNCName,
-	'xs:ID': isValidID,
-	'xs:IDREF': isValidID,
-	'xs:ENTITY': isValidENTITY,
-
-	'xs:integer': isValidInteger,
-
-	'xs:yearMonthDuration': isValidYearMonthDuration,
-	'xs:dayTimeDuration': isValidDayTimeDuration,
-};
-
-export default dataTypeValidatorByName;
+	const validatorToType: Map<ValueType, (value: string) => boolean> = new Map([
+		[ValueType.XSANYSIMPLETYPE, isValidAnySimpleType],
```

Comment:
Lets also hoist this map to the module level. It is constant, no need to redeclare it